crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1321,143 +1345,29 @@ impl<'db> TypeInferenceBuilder<'db> {
 
     fn infer_assignment_definition(
         &mut self,
-        target: &ast::Expr,
+        target: TargetKind<'db>,
         value: &ast::Expr,
         name: &ast::ExprName,
-        kind: AssignmentKind,
         definition: Definition<'db>,
     ) {
         let expression = self.index.expression(value);
         let result = infer_expression_types(self.db, expression);
         self.extend(result);
 
         let value_ty = self.expression_ty(value);
+        let name_ast_id = name.scoped_ast_id(self.db, self.scope());
 
-        let target_ty = match kind {
-            AssignmentKind::Sequence => self.infer_sequence_unpacking(target, value_ty, name),
-            AssignmentKind::Name => value_ty,
+        let target_ty = match target {
+            TargetKind::Sequence(unpack) => {
+                let unpacked = infer_unpack_types(self.db, unpack);
+                self.diagnostics.extend(unpacked.diagnostics());
```

Comment:
I guess the quick-fix approach to the duplicate diagnostics would be to track on `self` which unpacks we've queried already, and if we've seen it already, don't merge the diagnostics.

But I think it makes sense to see if accumulators can work, since that's a more general approach that doesn't require us to manually track when we might call a sub-query more than once.

Reply:
Yeah, I suggest not to spend more time on deduplicating diagnostics because I plan to look into this soon anyway